identity-wallet/src/state/reducers/dev_mode.rs

```diff
@@ -38,20 +42,104 @@ lazy_static! {
     );
 }
 
-pub async fn set_dev_mode(state: AppState, action: Action) -> Result<AppState, AppError> {
-    if let Some(enabled) = listen::<SetDevMode>(action).map(|payload| payload.enabled) {
-        return Ok(AppState {
-            dev_mode_enabled: enabled,
-            current_user_prompt: None,
-            ..state
-        });
+pub(super) const PASSWORD: &str = "sup3rSecr3t";
+
+pub async fn load_dev_profile(state: AppState, action: Action) -> Result<AppState, AppError> {
+    info!("Load dev profile: {:?}", action);
+
+    if let Some(dev_profile) = listen::<DevProfile>(action) {
+        // All dev profiles need to use the const PASSWORD so it can automatically unlock storage.
+        match dev_profile.profile {
+            ProfileType::Ferris => return load_ferris_profile().await,
+            ProfileType::Dragon => return load_dragon_profile(state, dev_profile).await,
+        }
     }
+
+    Ok(state)
+}
+
+pub async fn toggle_dev_settings(mut state: AppState, _action: Action) -> Result<AppState, AppError> {
+    info!("Toggle dev settings");
+
+    if state.dev_mode != DevMode::Off {
+        state.dev_mode = DevMode::Off;
+    } else {
+        // We don't preserve if user had autologin enabled
+        // So we just put it back to default (reload profile if you want to enable autologin again)
```

Comment:
If I follow the logic here, it would mean that:

1. A tester wants to try out a few things in a blank app and turns on `dev_mode`
2. Loads any dev profile to get some data to play around with (it doesn't matter if it's Ferris or Dragon)
3. The tester turns off `dev_mode` again to inspect how the app would look like to a regular user
4. When the app is now closed, you're locked out because: dev profile password is required + DevMode is not OnWithAutologin

Is this correct?

Reply:
Yes, if you want to enable it again you need to load dev profile

Reply:
Alright 👍 As long as we're aware of it, it's fine. We don't have to cover all edge cases at this point.